### PR TITLE
add documentation for PyCharm Plugin

### PR DESCRIPTION
**Pull request by koxudaxi**
## Change Summary
The PR adds documentation for PyCharm Plugin.

## Related issue number

#650

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.rst` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by dmontagu** [comment]
@koxudaxi Let me offer the following wordsmithing suggestions:
```
**plays nicely with your IDE/linter/brain**
    because *pydantic* data structures are just instances of classes you define;
    :ref:`auto-completion <pycharm_plugin>`, linting, :ref:`mypy <usage_mypy>` and your intuition
    should all work properly with your validated data.
```
(It seems awkward to me to call out PyCharm specifically this early in the docs, but I think it's okay to make a reference. I'd defer to @samuelcolvin though.)

```
.. _pycharm_plugin:

PyCharm Plugin
--------------

While pydantic will work well with any IDE out of the box, a PyCharm plugin
offering improved pydantic integration is available on the JetBrains Plugins Repository.
You can install the plugin for free from the plugin marketplace
(PyCharm's Preferences -> Plugin -> Marketplace).

The plugin currently supports the following features:

* For ``pydantic.BaseModel.__init__``:

     * Inspection
     * Autocompletion
     * Type-checking

* For fields of ``pydantic.BaseModel``:

    * Refactor-renaming fields updates ``__init__`` calls, and affects sub- and super-classes
    * Refactor-renaming ``__init__`` keyword arguments updates field names, and affects sub- and super-classes

More information can be found on the
`official plugin page <https://plugins.jetbrains.com/plugin/12861-pydantic>`_
and `Github repository <https://github.com/koxudaxi/pydantic-pycharm-plugin>`_.
```
Feel free to reject if you dislike/disagree with any of the changes.

**Comment by koxudaxi** [comment]
@dmontagu 
Thank you reviewing my document.
I feel great about your sentence.
I have merged only the section of `PyCharm Plugin`.

@samuelcolvin 
if you want to change the introduction section which is written dmontagu. (`**plays nicely with you`...) , then I merge one too.

File: docs/index.rst [hunk 1]
```diff
@@ -1199,6 +1199,34 @@ You'll need to have **python 3.6** or **3.7**, **virtualenv**, **git**, and **ma
...
 
+.. _pycharm_plugin:
+
+PyCharm Plugin
+--------------
+
+While pydantic will work well with any IDE out of the box, a PyCharm plugin
+offering improved pydantic integration is available on the JetBrains Plugins Repository.
+You can install the plugin for free from the plugin marketplace
+(PyCharm's Preferences -> Plugin -> Marketplace).
```
Comment (samuelcolvin) [inline docs/index.rst]:
```suggestion
(PyCharm's Preferences -> Plugin -> Marketplace -> search "pydantic").
```
I guess it's obvious, but still better to be explicit.

File: docs/index.rst [hunk 2]
```diff
@@ -51,7 +51,7 @@ So *pydantic* uses some cool new language feature, but why should I actually go
 
 **plays nicely with your IDE/linter/brain**
     because *pydantic* data structures are just instances of classes you define; auto-completion, linting,
-    :ref:`mypy <usage_mypy>` and your intuition should all work properly with your validated data.
+    :ref:`mypy <usage_mypy>`, :ref:`PyCharm Plugin <pycharm_plugin>` and your intuition should all work properly with your validated data.
```
Comment (samuelcolvin) [inline docs/index.rst]:
```suggestion
    :ref:`mypy <usage_mypy>`, IDEs (especially :ref:`PyCharm <pycharm_plugin>`) and your intuition should all work properly with your validated data.
```

Or something like that?

File: docs/index.rst [hunk 3]
```diff
@@ -1199,6 +1199,34 @@ You'll need to have **python 3.6** or **3.7**, **virtualenv**, **git**, and **ma
 **tl;dr**: use ``make format`` to fix formatting, ``make`` to run tests and linting & ``make docs``
 to build docs.
 
+.. _pycharm_plugin:
+
+PyCharm Plugin
+--------------
+
+While pydantic will work well with any IDE out of the box, a PyCharm plugin
+offering improved pydantic integration is available on the JetBrains Plugins Repository.
```
Comment (samuelcolvin) [inline docs/index.rst]:
```suggestion
offering improved pydantic integration is available on the JetBrains Plugins Repository for PyCharm.
```

**Comment by koxudaxi** [comment]
@samuelcolvin 
Thank you. I have committed!!

**Comment by samuelcolvin** [comment]
thank you very much.
